compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -261,6 +266,221 @@ static CodeGenPipeline getTensorCorePipeline(Type elementType) {
+    // Only single combiner operations are supported for now.
+    SmallVector<Operation *> combinerOps;
+    if (matchReduction(op.getRegionOutputArgs(), index, combinerOps) &&
+        combinerOps.size() == 1) {
+      if (foundSingleReductionOutput)
+        return failure();
+      foundSingleReductionOutput = true;
+      continue;
+    }
+    if (!op.getMatchingIndexingMap(&initOpOperand).isIdentity())
+      return failure();
+  }
+  if (!foundSingleReductionOutput)
+    return failure();
+
+  SmallVector<int64_t> workgroupTileSizes(op.getNumLoops(), 0);
+  SmallVector<int64_t> threadTileSizes(op.getNumLoops(), 0);
+  SmallVector<int64_t> partialReductionTileSizes(op.getNumLoops(), 0);
+  SmallVector<int64_t> threadCounts(op.getNumLoops(), 1);
+  SmallVector<int64_t> subgroupCounts(op.getNumLoops(), 1);
+  SmallVector<int64_t> mapping(op.getNumLoops());
+  std::iota(mapping.begin(), mapping.end(), 0);
+
+  int64_t lastReductionDim = reductionDims.back();
+
+  // Set the workgroup size to 1 for all parallel dimensions.
+  for (int64_t dim : parallelDims) {
+    workgroupTileSizes[dim] = 1;
+  }
+
+  // Set the threadTileSizes to 1 and partialReductionTileSizes to 1 for all
+  // reduction dimensions.
+  // We will later modify the threadTileSizes and partialReductionTileSizes for
+  // the last reduction dimension.
+  for (int64_t dim : reductionDims) {
+    threadTileSizes[dim] = 1;
+    partialReductionTileSizes[dim] = 1;
+  }
+
+  int64_t reductionSize = 1;
+  for (int64_t dim : reductionDims)
+    reductionSize *= bounds[dim];
+
+  int64_t subgroupSize = 0;
+  for (int s : wgp.getSubgroupSizeChoices().asArrayRef()) {
+    if (reductionSize % s == 0) {
+      subgroupSize = s;
+      break;
+    }
+  }
+
+  if (subgroupSize == 0)
+    return failure();
+
+  auto bitWidth = getBitWidth(op);
+  if (failed(bitWidth)) {
+    return failure();
+  }
+
+  const std::optional<int64_t> maxLoadBits = wgp.getMaxLoadInstructionBits();
+  const unsigned largestLoadSizeInBits =
+      maxLoadBits.has_value() ? *maxLoadBits : 128;
+
+  int64_t lastDimReductionSize = bounds[reductionDims.back()];
+
+  // The number of vector loads per thread.
+  int64_t threadLoads = largestLoadSizeInBits / *bitWidth;
+
+  // TODO: This is a temporary heuristic to ensure that the last dimension is
+  // divisible by the number of vector loads per thread.
+  if (lastDimReductionSize % threadLoads != 0)
+    return failure();
+
+  // The partial reduction would be no. of lanes (i.e., subgroup size) * #vector
+  // loads.
+  int64_t partialReductionSize = subgroupSize * threadLoads;
+
+  while (lastDimReductionSize % partialReductionSize != 0)
+    partialReductionSize >>= 1;
+
+  int64_t threadBasis = partialReductionSize / threadLoads;
+
+  partialReductionTileSizes[lastReductionDim] = partialReductionSize;
+  threadTileSizes[lastReductionDim] = threadLoads;
+  threadCounts[lastReductionDim] = threadBasis;
+
+  auto context = op.getContext();
```

Comment:
```suggestion
  MLIRContext* context = op.getContext();
```
See https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable